docs/inference_protocols.md

```diff
@@ -111,3 +111,42 @@ endpoints for Triton. These endpoints use the C API to communicate
 with the core of Triton. The primary source files for the endpoints
 are [grpc_server.cc](../src/servers/grpc_server.cc) and
 [http_server.cc](../src/servers/http_server.cc).
+
+## Java API (pre-release)
+
+The Triton Inference Server uses [Java CPP](https://github.com/bytedeco/javacpp)
+to create bindings around Tritonserver to create Java API. It requires the user
+to link a pre-built library to their application using Maven.   
+
+The API is documented in 
+[tritonserver.java](https://github.com/bytedeco/javacpp-presets/blob/master/tritonserver/src/gen/java/org/bytedeco/tritonserver/global/tritonserver.java).
+Alternatively, the user can refer to the web version [API docs](http://bytedeco.org/javacpp-presets/tritonserver/apidocs/)
+generated from `tritonserver.java`.
+A simple example using the Java API can be found in
+[Samples folder](https://github.com/bytedeco/javacpp-presets/tree/master/tritonserver/samples)
+which includes `Simple.java` which is similar to `simple.cc`. Please refer to
+[sample usage documentation](https://github.com/bytedeco/javacpp-presets/tree/master/tritonserver#sample-usage)
+to learn about how to build and run `Simple.java`.
+
+### Java API setup instructions
+
+The current snapshot is based on Triton container version `21.12`. To setup your
+enviroment with Triton Java API, please follow the following steps:
+1. First run Docker container:
+```
+ $ docker run -it --gpus=all -v ${pwd}:/workspace nvcr.io/nvidia/tritonserver:21.12-py3 bash
+```
+2. Then install `java sdk` and `maven`:
+```
+ $ apt update && apt install -y openjdk-11-jdk
+ $ wget https://archive.apache.org/dist/maven/maven-3/3.8.4/binaries/apache-maven-3.8.4-bin.tar.gz
+ $ tar zxvf apache-maven-3.8.4-bin.tar.gz
+ $ export PATH=/opt/tritonserver/apache-maven-3.8.4/bin:$PATH
+```
+3. Maven requires a `pom.xml` file to compile. The pre-built snapshots are hosted
```

Comment:
For this, related to your [comment in the simple test PR](https://github.com/triton-inference-server/server/pull/3771#pullrequestreview-849843641), we should decide whether we:

1. Update the pom.xml in javacpp-presets to work with the provided instructions.
2. Provide the pom.xml for the simple example (e.g. linking to the test pom.xml), so use can base their poms off it.

I think we may want to wait until we talk to Jack, since the current instructions did not work for us. It's helpful that you write which snapshot to use, but users may run into the same issue as we did. It'd be great to include the sample working pom to avoid that.

Reply:
Great updates on building with the current pom. Approving. These build instructions are clear and should work for the user, including if they use the linked samples pom.xml for simple (as an example).